__tests__/app.test.js

```diff
@@ -192,10 +192,123 @@ describe('PATCH /api/articles', () => {
+        .expect(200)
+        .then(({body}) => {
+            const { articles } = body
+            expect(articles).toBeSortedBy('created_at', {descending: true})
+        })
+    });
+    it('should return a 400 error when sort_By is not an existing validRow', () => {
+        return request(app)
+        .get('/api/articles?sort_by=notcollumn')
         .expect(400)
         .then(({body}) => {
-            expect(body).toEqual({msg : "bad request"})
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    it('should return a 400 error when sort_Order is not an existing validOrder', () => {
+        return request(app)
+        .get('/api/articles?order=gdfgf')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    it('returns all objects sorted by the provided column name', () => {
+        return request(app)
+        .get('/api/articles?sort_by=article_id')
+        .expect(200)
+        .then(({ body })=>{
+            const { articles } = body
+            expect(articles).toBeSortedBy('article_id', {descending: true})
+        })
+    })
+    it('returns an array of articles ordered by by descending order', () => {
+        return request(app).get(`/api/articles/?order=DESC`)
+        .expect(200)
+        .then(({body}) => {
+            const { articles } = body
+            expect(articles.length).toBe(12)
+            expect(articles).toBeSorted({descending: true})
+        })
+    });
+    it('returns array of articles sorted by descending order with specified topic', () => {
+        const query = '?topic=cats'
+        return request(app).get(`/api/articles${query}`)
+        .expect(200)
+        .then(({body}) => {
+            const { articles } = body
+            expect(articles.length).toBe(1)
+            expect(articles).toEqual([
+                {
+                    article_id:5,
+                    title: "UNCOVERED: catspiracy to bring down democracy",
+                    topic: "cats",
+                    author: "rogersop",
+                    body: "Bastet walks amongst us, and the cats are taking arms!",
+                    created_at: expect.any(String),
+                    votes: 0,
+                    comment_count: 2,
+                  }
+            ])
         })
     });
+    it('should return a 404 error when topic is not an existing column name', () => {
+        const query = '?topic=music'
+        return request(app)
+        .get(`/api/articles${query}`)
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe(`music not found`)
+        })
+    })
+    it('should return a 200 status with no articles found when topic is an existing column name', () => {
+        const query = '?topic=paper'
+        return request(app)
+        .get(`/api/articles${query}`)
+        .expect(200)
+        .then(({body}) => {
+            expect(body.msg).toBe('No articles found')
```

Comment:
Because this is successfully (200), you'd expect a response not a message i.e. an empty array.

This will break the frontend at the moment.
You can check on the front for an empty array, then display a message.